index.js

```diff
@@ -558,7 +558,7 @@ function buildArray (location) {
 
   functionCode += `
     if (!Array.isArray(obj)) {
-      throw new TypeError(\`The value '$\{obj}' does not match schema definition.\`)
+      throw new TypeError(\`The value '$\{JSON.stringify(obj)}' does not match schema definition.\`)
```

Comment:
I dont like it. Imagine that obj is a deeply nested Object. We would then JSON.stringify the whole object, just to say, that it is invalid.

Reply:
Indeed, how about we truncate the stringified object like this?
```
const obj = { fieldA: 'valueA', fieldB: 123, fieldC: 'valueC', fieldD: [1, 2, 3, 4, 5] };
const str = JSON.stringify(obj);
if (str.length > 24) {
    throw new TypeError(`The value '${str.slice(0, 23)}...' does not match schema definition.`);
}
throw new TypeError(`The value '${str}' does not match schema definition.`);
```

I'm not sure how to do this efficiently/neatly

Reply:
Its the same issue. I would probably prefer following error message:

throw new TypeError(`The value of ${key} does not match schema definition.`);

To determine key, you would need to check if you can derive the key from the location object.

Reply:
The major problem is `object` can be recursively reference itself and `JSON.stringify` do not deal with this situation.
Sometimes it is not the fault of developer but the library it using.

Referencing the `location` would be a better choice.

Reply:
You can put a location data here.

Reply:
From what I see, we could display `jsonPointer` as key?

Reply:
schemaId is probably not relevant  as it is just a UUID generated at runtime. So it doesnt help to track down the error. 

jsonPointer is helpful. So I would prefer jsonPointer.

Reply:
I've added 8a6b27f3ccc1bc49f97e22e27daa549ef198e076 to display `jsonPointer` instead.
https://github.com/fastify/fast-json-stringify/blob/8a6b27f3ccc1bc49f97e22e27daa549ef198e076/index.js#L560-L562